src/App.js

```diff
@@ -80,12 +88,12 @@ class App extends Component {
             {this.state.favorites ? '' : <button onClick={() => this.takeMeToFav()}> Favorites</button>}
           </NavLink>
         </nav>
-        <Route exact path='/' render={() => this.state.characters ? <CharacterContainter characters={this.state.characters} fav={this.addToFavorite} /> : ''} />
+        <Route exact path='/' render={() => this.state.characters ? <CharacterContainter characters={this.state.characters} fav={this.addToFavorite} hideFav={this.takeMeToFav} /> : ''} />
         <Route exact path='/fav' render={() => <FavCharacters favorites={this.state.favCharacters} remove={this.removeFromFavorites} />} />
         <Route exact path='/character/:id' render={({ match }) => {
           return <CharacterPage characterPage={match.params.id} />
         }} />
-        <Route exact path='/' render={() => <Pagination characters={this.state.characters} setCharacters={this.fetchCharacters} currentPage={this.state.currentPage} next={this.nextPage} prev={this.previousPage} />} />
+        <Route exact path='/' render={() => <Pagination characters={this.state.characters} setCharacters={this.fetchCharacters} currentPage={this.state.currentPage} next={this.nextPage} prev={this.previousPage} changePage={this.changePage} />} />
```

Comment:
Nice work with your Routes here. If you havent already and plan on adding an error 404 page, you can create a route for it ("/404"), and then redirect to it using a <Switch> and <Redirect> if the user inputs any path that doesnt currently exist.